scripts/ci/validatecode.ps1

```diff
@@ -0,0 +1,78 @@
+<#
+.SYNOPSIS
+    Validates the code to check for common patterns and usage that shouldn't be
+    checked in.
+.DESCRIPTION
+    This currently checks:
+
+    - That Boo.Lang isn't used anywhere in the code. This is an autocomplete option
+      that occurs when using Lists and other collections - the right thing to do
+      is to use System.Collections
+
+    Returns 0 if there are no issues, non-zero if there are.
+.PARAMETER Directory
+    The directory containing the code to validate. Only CSharp files
+.EXAMPLE
+    .\validatecode.ps1 -Directory c:\path\to\MRTK\Assets
+#>
+param(
+    [string]$Directory
+)
+
+function CheckBooLang(
+    [string]$FileName,
+    [string[]]$FileContent,
+    [int]$LineNumber
+) {
+    <#
+    .SYNOPSIS
+        Checks if the given file (at the given line number) contains a reference to Boo.Lang
+        Returns true if such a reference exists.
+    #>
+    if ($FileContent[$LineNumber] -match "^using Boo\.Lang;") {
+        Write-Host "An instance of Boo.Lang was found in $FileName at line $LineNumber "
+        Write-Host "Use System.Collections instead."
+        return $true;
+    }
+    return $false
+}
+
+if (-not $Directory) {
```

Comment:
Can you not make this a mandatory parameter?
``` PowerShell
[Parameter(Mandatory=$true)]
[string]$Directory
```

Reply:
Ah, didn't know, thanks for the pointer